### PR TITLE
box: fix `fselect()` behavior on binary data

### LINKED ISSUE #7040: fselect fails while parsing UUID in binary format
Tarantool 2.10.0-beta2-229-g1b1ddd226

Steps to reproduce:

```lua
tarantool> box.schema.create_space('test'):create_index('pk')
tarantool> box.space.test:format({{name='id', type='unsigned'}, {name='uuid'}})
tarantool> box.space.test:replace{1, require('uuid').bin()}
---
- [1, !!binary rmw8FcPVLUmu/NUcJBoBmg==]
...

tarantool> box.space.test:select()
---
- - [1, !!binary rmw8FcPVLUmu/NUcJBoBmg==]
...

tarantool> box.space.test:fselect()
---
- error: 'builtin/box/schema.lua:2060: bad argument #2 to ''max'' (number expected,
    got nil)'
...
```

### PR DESCRIPTION
**Pull request by Gumix**
Currently it throws an error when encounter binary data,
use Base64 encoding instead.

Closes #7040

### CODE REVIEW

File: test/engine/select.result
Comment (locker) [inline test/engine/select.result:3748]: IMO dumping binary data encoded in base64 without any specification looks confusing. Maybe, add `!!binary` tag, as we do in Console/YaML? Come to think of it, is there any point altogether in printing binary data to the console? Maybe, just print something like `<binary>` instead?

File: src/box/lua/schema.lua
Comment (locker) [inline src/box/lua/schema.lua:2084]: What happens if binary data comprises a valid, but non-printable UTF8 string? Maybe, we should mask such strings, too, as we do in Console/YaML:

https://github.com/tarantool/tarantool/blob/5557c47a39f36c1d00291050854fa9cbf1382ae1/third_party/lua-yaml/lyaml.cc#L667

**Review by locker — COMMENTED** [review]
Please don't set the `full-ci` label until you get all approvals, unless you deliberately want to run tests on all platforms (e.g. patch platform-dependent code), so as not to overload CI.

File: test/engine/select.result
Comment (Gumix) [reply test/engine/select.result:3746]: This issue is known: https://github.com/tarantool/tarantool/blob/master/test/engine/gh-6165-multibyte-fselect.result#L95

File: src/box/lua/schema.lua
Comment (Gumix) [reply src/box/lua/schema.lua:2084]: Valid non-printable chars are escaped by `json.encode` above.

File: test/engine/select.result
Comment (Gumix) [reply test/engine/select.result]: Done

**Review by locker — APPROVED** [review]

**Review by alyapunov — APPROVED** [review]

**Pull request merged**
